configs/components/ruby-2.7.1.rb

```diff
@@ -198,6 +198,9 @@
     end
   elsif platform.is_windows?
     rbconfig_changes["CC"] = "x86_64-w64-mingw32-gcc"
+  elsif platform.name =~ /el-7-x86_64/
+    # EL 7 GCC (4.8.x) does not support -fstack-protector-strong
+    rbconfig_changes["LDFLAGS"] = "-L. -Wl,-rpath=/opt/puppetlabs/puppet/lib -fstack-protector -rdynamic -Wl,-export-dynamic -L/opt/puppetlabs/puppet/lib"
```

Comment:
I wonder if we can avoid overriding LDFLAGS (especially since it's project-based, pdk should use `/opt/puppetlabs/pdk` instead of `/opt/puppetlabs/puppet`). Maybe we can get away with setting `rbconfig_changes["CC"] = "/usr/bin/gcc"`, so gems with native extensions use the system Ruby which does support the flag.